client/src/rpc_client.rs

```diff
@@ -4045,6 +4045,14 @@ impl RpcClient {
         // lesser evil.
         tokio::task::block_in_place(move || self.runtime.as_ref().expect("runtime").block_on(f))
     }
+
+    pub(crate) fn clone_inner_client(&self) -> Arc<nonblocking::rpc_client::RpcClient> {
```

Comment:
Nit: maybe rename this to get_inner_client or get_async_client - the current name could be taken to mean we're actually cloning/copying the inner async client. Other than that, looks good.

Reply:
I went with `get_inner_client` -- I had a hard time with this name :sweat_smile: I had it return a `&Arc<...>` and then we clone from there.  Cloning the Arc in the getter is more work than should be necessary. Let me know what you think.

Reply:
LGTM